sklearn/metrics/_scorer.py

```diff
@@ -65,19 +66,21 @@
 
 from ..utils.multiclass import type_of_target
 from ..base import is_regressor
+from ..utils._response import _get_response_values
 from ..utils._param_validation import validate_params
 
 
-def _cached_call(cache, estimator, method, *args, **kwargs):
+def _cached_call(cache, estimator, *args, **kwargs):
     """Call estimator with method and args and kwargs."""
     if cache is None:
-        return getattr(estimator, method)(*args, **kwargs)
+        return _get_response_values(estimator, *args, **kwargs)
 
+    response_method = kwargs["response_method"]
```

Comment:
I find it weird to use kwargs here. Couldn't you keep it a regular (positional) arg of `_cached_call` as it was before if it's always required?

Alternatively make it a named kwarg, outside of the `kwargs` dict but with a default value that makes sense.

Reply:
Indeed, I just reused the past code but it would be clearer to keep the `response_method` as a mandatory arg.